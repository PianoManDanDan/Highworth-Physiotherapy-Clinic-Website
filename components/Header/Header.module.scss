@use '../../styles/mixins/media-query' as *;

.header {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
};

.content-wrapper {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  margin: 1.5rem 0 0.75rem;
};

.logo {
  margin: 0 0.75rem;
};

.text-wrapper {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
};

.title {
  margin: 0.5rem;
  font-size: 2.5rem;
  &:visited, &:active {
    color: inherit;
  };
};

.contact {
  display: flex;
  font-style: normal;

  &__number {
    display: none;
    @include media-query(small) {
      display: block;
      margin-left: 0.5rem;
    };
  };
};
