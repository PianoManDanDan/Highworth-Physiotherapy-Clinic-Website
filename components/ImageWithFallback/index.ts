export * from './ImageWithFallback';
export * from './ImageWithFallbackProps';
