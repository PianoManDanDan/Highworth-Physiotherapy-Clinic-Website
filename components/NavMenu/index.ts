export * from './NavMenu';
export * from './subcomponents';