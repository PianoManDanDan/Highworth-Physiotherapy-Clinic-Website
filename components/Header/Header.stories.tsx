import { StoryObj, Meta } from '@storybook/react';
import { ThemeProvider } from 'next-themes';
import { Header } from 'components/Header';

export default { component: Header } as Meta;

export const header: StoryObj<typeof Header> = {
  render: () => (
    <ThemeProvider disableTransitionOnChange>
      <Header />
    </ThemeProvider>
  ),
  parameters: {
    controls: { hideNoControlsWarning: true },
  }
};
