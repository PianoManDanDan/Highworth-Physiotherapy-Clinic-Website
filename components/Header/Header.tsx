import React from 'react';
import Image from 'next/image';
import Link from 'next/link';
import { Phone } from 'react-feather';
import { NavMenu, NavMenuItem } from '../NavMenu';
import styles from './Header.module.scss';

export const Header: React.FC<{}> = () => {
  return (
    <header className={styles.header}>
      <div className={styles['content-wrapper']}>
        <Link href='/'>
          <a className={styles.logo}>
            <Image
              src='/assets/logos/hpc.png'
              alt='Highworth Physiotherapy Clinic'
              height={128}
              width={30}
              layout='intrinsic'
            />
          </a>
        </Link>
        <div className={styles['text-wrapper']}>
          <Link href='/'>
            <a>
              <h1 className={styles.title}>Highworth Physiotherapy Clinic</h1>
            </a>
          </Link>
          <address>
            <h2>
              <a
                className={styles.contact}
                href="tel:+441793763814"
              >
                <Phone />
                <span className={styles.contact__number}>01793 763814</span>
              </a>
            </h2>
          </address>
        </div>
      </div>
      <NavMenu>
        <NavMenuItem text='Home' href='/' />
        <NavMenuItem text='Facilities' href='/facilities' />
        <NavMenuItem text='Physiotherapy' href='/physiotherapy' />
        <NavMenuItem text='Animals' href='/animals' />
        <NavMenuItem text='Meet the Team' href='/meet-the-team' />
        <NavMenuItem text="FAQ's" href='/faq' />
        <NavMenuItem text='Contact Us' href='/contact' />
      </NavMenu>
    </header>
  );
};

export default Header;
