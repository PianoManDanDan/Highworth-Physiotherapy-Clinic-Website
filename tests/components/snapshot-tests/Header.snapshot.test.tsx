import React from 'react';
import { render } from '@testing-library/react';
import { Header } from '../../../components/Header';

describe('Header Component snapshot test', () => {
  test('Header should render the text \'Highworth Physiotherapy Clinic\', the logo, a telephone link and the NavMenu with links to all pages in the app', () => {
    // Arrange
    const { asFragment } = render(
      <Header />
    );

    // Act
    const renderedHeader = asFragment();

    // Assert
    expect(renderedHeader).toMatchSnapshot();
  });
});
