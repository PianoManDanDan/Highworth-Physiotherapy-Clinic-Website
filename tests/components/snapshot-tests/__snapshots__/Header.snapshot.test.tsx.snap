// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component snapshot test Header should render the text 'Highworth Physiotherapy Clinic', the logo, a telephone link and the NavMenu with links to all pages in the app 1`] = `
<DocumentFragment>
  <header>
    <div>
      <a
        href="/"
      >
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzAiIGhlaWdodD0iMTI4IiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEuMSIvPg=="
              style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
            />
          </div>
          <noscript />
          <img
            alt="Highworth Physiotherapy Clinic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </div>
      </a>
      <div>
        <a
          href="/"
        >
          <h1>
            Highworth Physiotherapy Clinic
          </h1>
        </a>
        <address>
          <h2>
            <a
              href="tel:+441793763814"
            >
              <svg
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M22 16.92v3a2 2 0 0 1-2.18 2 19.79 19.79 0 0 1-8.63-3.07 19.5 19.5 0 0 1-6-6 19.79 19.79 0 0 1-3.07-8.67A2 2 0 0 1 4.11 2h3a2 2 0 0 1 2 1.72 12.84 12.84 0 0 0 .7 2.81 2 2 0 0 1-.45 2.11L8.09 9.91a16 16 0 0 0 6 6l1.27-1.27a2 2 0 0 1 2.11-.45 12.84 12.84 0 0 0 2.81.7A2 2 0 0 1 22 16.92z"
                />
              </svg>
              <span>
                01793 763814
              </span>
            </a>
          </h2>
        </address>
      </div>
    </div>
    <nav>
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/facilities"
      >
        Facilities
      </a>
      <a
        href="/physiotherapy"
      >
        Physiotherapy
      </a>
      <a
        href="/animals"
      >
        Animals
      </a>
      <a
        href="/meet-the-team"
      >
        Meet the Team
      </a>
      <a
        href="/faq"
      >
        FAQ's
      </a>
      <a
        href="/contact"
      >
        Contact Us
      </a>
    </nav>
  </header>
</DocumentFragment>
`;
